test-suite/src/test/scala/org/scalajs/testsuite/compiler/OptimizerTest.scala

```diff
@@ -81,6 +81,64 @@ object OptimizerTest extends JasmineTest {
 
   }
 
+  describe("String_+ constant folding") {
+    it("must fold two constant string to the correct value") {
+      expect("I am " + "constant").toEqual("I am constant")
+    }
+
+    it("must fold the empty string when associated with a string") {
+      val number: String = "1" + (1.4f).toString.substring(1)
+      expect(number + "").toEqual("1.399999976158142")
+      expect("" + number).toEqual("1.399999976158142")
+    }
+
+    it("must fold 1.4f and a stringLit to the correct value") {
+      expect(1.4f + "hello").toEqual("1.399999976158142hello")
+      expect("hello" + 1.4f).toEqual("hello1.399999976158142")
+    }
+
+    it("must fold cascading String + correctly") {
+      def someString(a: Int) = "awesome! " + a + "/10"
```

Comment:
`@noinline` if you intend this not to be inlined (maybe one day we will automatically inline small expressions).